src/NServiceBus.Core/Transports/TransportSeam.cs

```diff
@@ -26,14 +26,16 @@ public void Configure(ReceiveSettings[] receivers)
             this.receivers = receivers;
         }
 
+        // The dependency in IServiceProvider ensures that the TransportInfrastructure can't be resolved too early.
+        public TransportInfrastructure GetTransportInfrastructure(IServiceProvider _) => TransportInfrastructure;
+
+        public ITransportAddressResolver GetTransportAddressResolver(IServiceProvider sp) => sp.GetRequiredService<ITransportAddressResolver>();
```

Comment:
again, not sure if it's worth it, but it makes component dependencies at least a bit more visible and navigatable. But there are still some things that resolve it straight from DI (the MDPS subscription features because those are still features and not components)